libvirt.go

```diff
@@ -59,6 +59,9 @@ type Libvirt struct {
 	r    *bufio.Reader
 	w    *bufio.Writer
 	mu   *sync.Mutex
+	// closed after cleanup complete following the underlying connection to
+	// libvirt being disconnected.
+	socketCleanupDone chan struct{}
```

Comment:
that name is because in some yet to be submitted code, all the socket stuff is moved to a socket object, and this channel only becomes the thing indicating it's done cleaning up a particular socket connection (of which you could have more than one on a single Libvirt object).  So I'm not sure shutdown is quite accurate.  Maybe disconnectDone ?

Reply:
`disconnected` works for me